grakn-engine/src/main/java/ai/grakn/engine/GraknEngineServer.java

```diff
@@ -160,18 +158,20 @@ private void loadAndUnlock(Lock lock) {
 
     /**
      * Check in with the properties file to decide which type of task manager should be started
-     * @param inMemoryQueue
-     * @param redisCountStorage
+     * and return the TaskManager
+     * @param inMemoryQueue         True if running in memory
      * @param jedisPool
      */
     private TaskManager startTaskManager(
             boolean inMemoryQueue,
-            RedisCountStorage redisCountStorage,
-            Pool<Jedis> jedisPool, LockProvider lockProvider) {
+            Pool<Jedis> jedisPool,
+            LockProvider lockProvider) {
         TaskManager taskManager;
         if (!inMemoryQueue) {
             taskManager = new RedisTaskManager(engineId, prop, jedisPool, factory, lockProvider, metricRegistry);
         } else  {
+            // Redis storage for counts, in the RedisTaskManager it's created in consumers
+            RedisCountStorage redisCountStorage = RedisCountStorage.create(jedisPool);
```

Comment:
Just seems a bit weird. I am wondering if we can just collapse everything into one task manager and just get rid of this two mode thing all together. Before hand it was not possible because the "distributed mode" required ZK and Kafka, but now both modes just need Redis and nothing else. Do you think it will be possible to just collapse everything into one manager?